src/textUtils.c

```diff
@@ -101,7 +116,40 @@ size_t str_formatTtl(uint64_t ttl, char* out, size_t outSize)
 	return strlen(out);
 }
 
+// returns length of the resulting string
 size_t str_formatMetadata(const uint8_t* metadataHash, size_t metadataHashSize, char* out, size_t outSize)
 {
 	return encode_hex(metadataHash, metadataHashSize, out, outSize);
 }
+
+// check if it is ASCII between 32 and 126
+void str_validateTextBuffer(const uint8_t* text, size_t textSize)
+{
+	ASSERT(textSize < BUFFER_SIZE_PARANOIA);
+
+	for (size_t i = 0; i < textSize; i++) {
+		VALIDATE(text[i] <= 126, ERR_INVALID_DATA);
+		VALIDATE(text[i] >= 32, ERR_INVALID_DATA);
+	}
+}
+
+#ifdef DEVEL
+
+// converts a text to bytes (suitable for CBORization) and validates if chars are allowed
+size_t str_textToBuffer(const char* text, uint8_t* buffer, size_t bufferSize)
```

Comment:
I would suggest separating the validation and "cborization" logic and do the validation explicitly wherever it makes sense. But I see that this helper function is used only for tests, so I would drop the char code validation from there completely - there's imho not a point in validating test fixtures if they would not be validated in the actual logic either.

And if anybody deems this function reusable in the future for actual logic, they should still do due diligence and implement/use whatever input validation makes sense for their use case (possibly this one, possibly other)

Reply:
Fixtures are now validated in precisely the same way as actual data, so I'm keeping the validation here. 

(Generally, I have messed up test fixtures many times, so I do feel some need to validate them whenever it is easy and cheap.)